tests/test_jira.py

```diff
@@ -0,0 +1,380 @@
+
+
+class TestJiraClient(unittest.TestCase):
+    def test___init__(self):
+        client = JiraClient(url='http://example.com', project='perceval',
+                            user='user', password='password',
+                            verify=False, cert=None, max_issues=100)
+
+        self.assertEqual(client.url, 'http://example.com')
+        self.assertEqual(client.project, 'perceval')
+        self.assertEqual(client.user, 'user')
+        self.assertEqual(client.password, 'password')
+        self.assertEqual(client.verify, False)
+        self.assertEqual(client.cert, None)
+        self.assertEqual(client.max_issues, 100)
+
+    @httpretty.activate
+    def test_get_issues(self):
+
+        from_date = str_to_datetime('2015-01-01')
+
+        requests = []
+
+        bodies_json = [read_file('data/jira/jira_issues_page_1.json'),
+                       read_file('data/jira/jira_issues_page_2.json')]
+        expected_json = [read_file('data/jira/jira_issues_expected_page_1.json'),
+                         read_file('data/jira/jira_issues_expected_page_2.json')]
+
+        def request_callback(method, uri, headers):
+            body = bodies_json.pop(0)
+            requests.append(httpretty.last_request())
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               JIRA_SEARCH_URL,
+                               responses=[httpretty.Response(body=request_callback) \
+                                          for _ in range(2)])
+
+        client = JiraClient(url='http://example.com', project='perceval',
+                            user='user', password='password',
+                            verify=False, cert=None, max_issues=2)
+
+        pages = [page for page in client.get_issues(from_date)]
+
+        expected_req_0 = {
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' project = perceval AND  updated > 1420070400000'],
+                            'maxResults': ['2'],
+                            'startAt': ['0']
+                        }
+        expected_req_1 = {
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' project = perceval AND  updated > 1420070400000'],
+                            'maxResults': ['2'],
+                            'startAt': ['2']
+                        }
+
+        self.assertEqual(len(pages), 2)
+
+        self.assertEqual(requests[0].method, 'GET')
+        self.assertRegex(requests[0].path, '/rest/api/2/search')
+        self.assertDictEqual(requests[0].querystring, expected_req_0)
+
+        self.assertEqual(requests[1].method, 'GET')
+        self.assertRegex(requests[1].path, '/rest/api/2/search')
+        self.assertDictEqual(requests[1].querystring, expected_req_1)
+
+        self.assertEqual(pages[0], expected_json[0])
```

Comment:
You don't have to compare the result with other files. You need to compare it with what you sent, that it is stored in `bodies_json` list.